Sources/BrowserServicesKit/Domain/AppVersion.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  AppVersion.swift
+//  DuckDuckGo
+//
+//  Copyright © 2017 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// 1. App version is moved to BSK and it's assumed Mac can use this struct without any changes.
+/// 2. Create AppVersion protocol in BSK and create struct that implements it in Core app called iOSAppVersion.swift. (APMobile)
+
+public struct AppVersion {
```

Comment:
Regardless of pending conversations wrt BSK I'm not sure I like this being in BSK since it's info so inherent to the client. But not sure I can substantiate that feeling, so if you are reading this Bartek I'm keen to see what you think

Reply:
Personally, I think it's fine. This library is intended to support making development of the clients easier/quicker and our clients have these keys so it seems reasonable to share this code imo.  

fwiw I had also started to do this kind of thing in the sync branches before I got seconded.  What I want to avoid is spaghetti code and creating additional protocols and wiring for the clients is just that - hence, I also support moving the network layer into this package as well, but I guess we'll want to discuss all that more widely.

Reply:
AppVersion is required for the remote messaging rules so it is possible of course to cherry pick the properties needed but I felt like this was the best approach.

Remote messaging framework uses AppVersion like the following:
- if AppVersion is X, do Y
- if AppVersion is > X, but < Y, do Z 
- etc